bundles/org.openhab.binding.ambientweather/src/main/java/org/openhab/binding/ambientweather/internal/handler/AmbientWeatherEventListener.java

```diff
@@ -0,0 +1,352 @@
+        logger.debug("Listener: Remove station handler from list: {}", handler.getThing().getUID());
+        handlers.remove(handler);
+    }
+
+    /*
+     * Send weather station information (station name and location) to the
+     * thing handler associated with the MAC address
+     */
+    private void sendStationInfoToHandler(String macAddress, String name, String location) {
+        AmbientWeatherStationHandler handler = getHandler(macAddress);
+        if (handler != null) {
+            handler.handleInfoEvent(macAddress, name, location);
+        }
+    }
+
+    /*
+     * Send an Ambient Weather data event to the station thing handler associated
+     * with the MAC address
+     */
+    private void sendWeatherDataToHandler(String macAddress, String jsonData) {
+        AmbientWeatherStationHandler handler = getHandler(macAddress);
+        if (handler != null) {
+            handler.handleWeatherDataEvent(jsonData);
+        }
+    }
+
+    private AmbientWeatherStationHandler getHandler(String macAddress) {
+        logger.debug("Listener: Search for MAC {} in handlers list with {} entries", macAddress, handlers.size());
+        for (Map.Entry<AmbientWeatherStationHandler, String> device : handlers.entrySet()) {
+            AmbientWeatherStationHandler handler = device.getKey();
+            String mac = device.getValue();
+            if (mac.equals(macAddress)) {
+                logger.debug("Listener: Found handler for {} with MAC {}", handler.getThing().getUID(), macAddress);
+                return handler;
+            }
+        }
+        logger.debug("Listener: No handler available for event for station with MAC {}", macAddress);
+        return null;
+    }
+
+    /*
+     * Start the event listener for the Ambient Weather real-time API
+     */
+    public void start(String applicationKey, String apiKey, Gson gson) {
+        logger.debug("Listener: Event listener starting");
+        this.applicationKey = applicationKey;
+        this.apiKey = apiKey;
+        this.gson = gson;
+        connectToService();
+    }
+
+    /*
+     * Stop the event listener for the Ambient Weather real-time API.
+     */
+    public void stop() {
+        logger.debug("Listener: Event listener stopping");
+        sendUnsubscribe();
+        disconnectFromService();
+        handlers.clear();
+    }
+
+    /*
+     * Initiate the connection to the Ambient Weather real-time API
+     */
+    private synchronized void connectToService() {
+        final String url = REALTIME_URL.replace("%APPKEY%", applicationKey);
+        try {
+            IO.Options options = new IO.Options();
+            options.forceNew = true;
+            options.transports = new String[] { "websocket" };
+            socket = IO.socket(url, options);
+        } catch (URISyntaxException e) {
+            logger.error("Listener: URISyntaxException getting IO socket: {}", e.getMessage());
```

Comment:
Please do not use logger error level. This is reserved for the framework. Thanks.

Reply:
Good spot! Can't believe I missed that. Fixed now.